src/components/Login/LoginForm.js

```diff
@@ -0,0 +1,65 @@
+import { useNavigate, useLocation } from "react-router-dom"
+import { Form, Text, Button, Input, Label } from "../Reusable"
+import { getSolidBtnBgColor, getSolidBtnTextColor } from "../../utils"
+import { useLogin, useTheme, useAuth } from '../../contexts'
+
+const LoginForm = () => {
+    const navigate = useNavigate()
+    const location = useLocation()
+    const { theme } = useTheme()
+    const { loginFormState, isFormInvalid, loginFormDispatch, showLoginAlert } = useLogin()
+    const { loginUser } = useAuth()
+    const from = location.state?.from?.pathname || '/videos'
```

Comment:
Appreciation: good use of optional chaining here

Reply:
thankyou